### PR TITLE
Statistics-- update

### PR DESCRIPTION
**Pull request by renukamanavalan**
**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: src/sonic-eventd/Makefile [hunk 1]
```diff
@@ -25,7 +25,8 @@ endif
 -include rsyslog_plugin/subdir.mk
 -include rsyslog_plugin_tests/subdir.mk
 
-all: sonic-eventd eventd-tests eventd-tool rsyslog-plugin rsyslog-plugin-tests
+# all: sonic-eventd eventd-tests eventd-tool rsyslog-plugin rsyslog-plugin-tests
+all: sonic-eventd eventd-tests eventd-tool rsyslog-plugin
```
Comment (zbud-msft) [inline src/sonic-eventd/Makefile]: Tests should pass

File: src/sonic-eventd/tests/main.cpp [hunk 2]
```diff
@@ -1,10 +1,97 @@
...
+#define INVALID_NAMESPACE  "invalid"
+
+bool g_is_redis_available = false;
+
+class SwsscommonEnvironment : public ::testing::Environment {
+public:
+    // Override this to define how to set up the environment.
+    void SetUp() override {
+        // by default , init should be false
+        cout<<"Default : isInit = "<<SonicDBConfig::isInit()<<endl;
```
Comment (zbud-msft) [inline src/sonic-eventd/tests/main.cpp:23]: Uses SWSS_LOG instead of cout

File: src/sonic-eventd/tests/main.cpp [hunk 3]
```diff
@@ -1,10 +1,97 @@
...
+
+        // Get this info handy
+        try 
+        {
+            DBConnector db("TEST_DB", 0, true);
+            g_is_redis_available = true;
+        }
+        catch (exception &e)
+        {
+            printf("Unable to get DB Connector, e=(%s)\n", e.what());
```
Comment (zbud-msft) [inline src/sonic-eventd/tests/main.cpp:83]: SWSS_LOG_ERROR

File: src/sonic-eventd/src/eventd.h [hunk 4]
```diff
@@ -42,6 +56,93 @@ class eventd_proxy
...
+        }
+
+        bool is_running()
+        {
+            return !m_shutdown;
+        }
+
+    private:
+        void _update_stats(int index, counters_t val) {
+            m_lst_counters[index] += val;
```
Comment (zbud-msft) [inline src/sonic-eventd/src/eventd.h]: Check for index out of bounds

File: src/sonic-eventd/src/eventd.cpp [hunk 5]
```diff
@@ -65,6 +95,169 @@ eventd_proxy::run()
...
+
+        try {
+            rc = event_receive(subs_handle, op);
+        }
+        catch (exception& e)
+        {
+            rc = -1;
+            stringstream ss;
+            ss << e.what();
+            SWSS_LOG_ERROR("Cache save event failed with %s",
```
Comment (zbud-msft) [inline src/sonic-eventd/src/eventd.cpp]: Shouldnt log message should be about event_receive failing

File: src/sonic-eventd/src/eventd.cpp [hunk 6]
```diff
@@ -65,6 +95,169 @@ eventd_proxy::run()
...
+     * But consider, that eventd shutdown is a critical shutdown as it would
+     * bring down all other features. Hence done only at system level shutdown,
+     * hence losing few messages in flight is acceptable. Any more complex code
+     * to handle is unwanted.
+     */
+
+    events_deinit_subscriber(subs_handle);
+    events_deinit_publisher(pub_handle);
+    m_shutdown = true;
+    return;
```
Comment (zbud-msft) [inline src/sonic-eventd/src/eventd.cpp]: No need for this return statement

File: src/sonic-eventd/src/eventd.h [hunk 7]
```diff
@@ -42,6 +56,93 @@ class eventd_proxy
...
+class stats_collector
+{
+    public:
+        stats_collector();
+
+        ~stats_collector() { stop(); }
+
+        int start();
+
+        void stop() {
```
Comment (zbud-msft) [inline src/sonic-eventd/src/eventd.h:70]: for all func definitions before this, you have '{' on new line

File: src/sonic-eventd/tests/eventd_ut.cpp [hunk 8]
```diff
@@ -229,12 +223,12 @@ void run_pub(void *mock_pub, const string wr_source, internal_events_lst_t &lst)
 void debug_on()
 {
     /* compile with -D DEBUG_TEST or add "#define DEBUG_TEST" to include. */
-#ifdef DEBUG_TEST
+// #ifdef DEBUG_TEST
```
Comment (zbud-msft) [inline src/sonic-eventd/tests/eventd_ut.cpp]: Delete commented out code

File: src/sonic-eventd/src/eventd.h [hunk 9]
```diff
@@ -42,6 +56,93 @@ class eventd_proxy
...
+    private:
+        void _update_stats(int index, counters_t val) {
+            m_lst_counters[index] += val;
+            m_updated = true;
+        }
+
+        void run_collector();
+
+        void run_writer();
+
```
Comment (zbud-msft) [inline src/sonic-eventd/src/eventd.h:139]: delete extra empty line

**Review by zbud-msft — COMMENTED** [review]
Why are we updating submodule, wpasupplicant?

**Comment by zbud-msft** [comment]
I think you can also remove sonic-swss-common submodule update, as master sonic-buildimage has update

File: src/sonic-eventd/Makefile (hunk 1 above)
Comment (renukamanavalan) [reply src/sonic-eventd/Makefile]: Good catch. I will do this.

File: src/sonic-eventd/tests/main.cpp (hunk 2 above)
Comment (renukamanavalan) [reply src/sonic-eventd/tests/main.cpp:23]: This is by design. For unit tests, we want the spew in stdout, so the build log will have it. 
BTW, this is the same code from unit test of swsscommon.

File: src/sonic-eventd/tests/main.cpp (hunk 3 above)
Comment (renukamanavalan) [reply src/sonic-eventd/tests/main.cpp:83]: Same as above --Not for unit tests.

File: src/sonic-eventd/src/eventd.h (hunk 4 above)
Comment (renukamanavalan) [reply src/sonic-eventd/src/eventd.h]: This is a private method, hence may not need. Yet it is a good idea, so as not to crash

Comment (renukamanavalan) [reply src/sonic-eventd/src/eventd.h]: Done

File: src/sonic-eventd/src/eventd.cpp (hunk 6 above)
Comment (renukamanavalan) [reply src/sonic-eventd/src/eventd.cpp]: Done

File: src/sonic-eventd/src/eventd.h (hunk 9 above)
Comment (renukamanavalan) [reply src/sonic-eventd/src/eventd.h:139]: Done

File: src/sonic-eventd/Makefile (hunk 1 above)
Comment (renukamanavalan) [reply src/sonic-eventd/Makefile]: done.

File: src/sonic-eventd/src/eventd.cpp (hunk 5 above)
Comment (renukamanavalan) [reply src/sonic-eventd/src/eventd.cpp]: Yes. Corrected. Thanks.

File: src/sonic-eventd/tests/eventd_ut.cpp (hunk 8 above)
Comment (renukamanavalan) [reply src/sonic-eventd/tests/eventd_ut.cpp]: This entire function is removed as redundant.

**Comment by zbud-msft** [comment]
You are updating submodule wpa-supplicant and there are conflicts with swss-common submodule as well

**Comment by renukamanavalan** [comment]
Incorrect submodule updates in this PR could not be reverted. Hence abandoning this PR.

**Pull request closed without merging**